### ISSUE TITLE
Invalid `@UnnecessaryLambda` patch suggestion

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#2518]
Consider the following code:
```java
import java.util.stream.IntStream;

class Example {
  void someLoopyCode() {
    for (int i : someIterable()) {
      // Do something.
    }
  }

  private Iterable<Integer> someIterable() {
    return () -> IntStream.range(0, 42).boxed().iterator();
  }
}
```

Compile the code as follows:
```sh
javac \
  -XDcompilePolicy=simple \
  -processorpath error_prone_core-2.9.0-with-dependencies.jar \
  '-Xplugin:ErrorProne -XepDisableAllChecks -Xep:UnnecessaryLambda:WARN' \
  Example.java
```

This yields the following warning:
```
Example.java:10: warning: [UnnecessaryLambda] Returning a lambda from a helper method or saving it in a constant is unnecessary; prefer to implement the functional interface method directly and use a method reference instead.
  private Iterable<Integer> someIterable() {
                            ^
    (see https://errorprone.info/bugpattern/UnnecessaryLambda)
  Did you mean 'for (int i : this::someIterable) {'?
1 warning
```

If this suggestion were accepted using `-XepPatchChecks:UnnecessaryLambda` the resulting code would not compile, producing a "_method reference not expected here_" error. Possible solutions:
1. Don't flag enhanced for-loop method calls.
2. Introduce an explicit cast (i.e., suggest `for (int i : (Iterable<Integer>) this::someIterable) {` instead).

**Comment by cushon** [comment]
https://github.com/google/error-prone/commit/93ca62e71a836684d2fe103883f37e7976b69f0d
